dask/blockwise.py

```diff
@@ -225,12 +267,58 @@ def _dict(self):
                 key_deps=key_deps,
                 non_blockwise_keys=non_blockwise_keys,
             )
+
+            if self.io_subgraph:
+                # This is an IO layer.
+                for k in dsk:
+                    io_key = (self.io_name,) + tuple([k[i] for i in range(1, len(k))])
+                    if io_key in dsk[k]:
+                        # Inject IO-function arguments into the blockwise graph
+                        # as a single (packed) tuple.
+                        io_item = self.io_subgraph.get(io_key)
+                        io_item = list(io_item[1:]) if len(io_item) > 1 else []
+                        new_task = [io_item if v == io_key else v for v in dsk[k]]
+                        dsk[k] = tuple(new_task)
+
+                # Clear IO "placeholder" dependencies
+                for k in key_deps:
+                    if k[0] == self.output:
+                        key_deps[k] = set()
+
             self._cached_dict = {
                 "dsk": dsk,
                 "basic_layer": BasicLayer(dsk, key_deps, non_blockwise_keys),
             }
         return self._cached_dict["dsk"]
 
+    def get_output_keys(self):
+
+        # TODO: Handle N-D Collections and more-complex
+        # tensor operations.
```

Comment:
Could you open up an issue with checkboxes to track follow-up work we'll want to do to ensure we're using using IO-subgraphs throughout the codebase

Reply:
I added #6791 (to track the use of IO subgraphs) and #6792 (to track improvements in this particular method) - Let me know if you had anything else in mind.  Also, feel free to modify those issues in any way you'd like :)